app/controllers/api/v1/application_controller.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class ApplicationController < ActionController::API
```

Comment:
We follow [JSON API](https://jsonapi.org/) standard as per our [convention](https://nimblehq.co/compass/development/code-conventions/#api-design). Based on your postman `users/create`, `oauth/token` API response, its is not aligining  with [JSON API](https://jsonapi.org/) standard.

For example `oauth/token` endpoint response should be like bellow as per  [JSON API](https://jsonapi.org/) standard 
![Screen Shot 2021-06-17 at 10 41 10 AM](https://user-images.githubusercontent.com/6076762/122327780-a6f3cf80-cf58-11eb-8220-019f1964f80c.png)

Also `users/create` endpont should return `201` success status code instead of `200`. For resource creation, it shound return `201`.

You can check our Nimble Servey API response on [postman](https://nimblehq.postman.co/workspace/Nimble~9daf6b25-882e-4ad0-8299-95529e2883ff/documentation/11835486-63105f38-c20f-493d-b4d8-873a18c8d8c7)

For referrence, you can check Nimble Servey API codebase([controller](https://github.com/nimblehq/nimble-survey-web/tree/development/app/controllers/api/v1), [concern](https://github.com/nimblehq/nimble-survey-web/tree/development/app/controllers/concerns/api/v1)) which also used Devise, doorkeeper, fast_jsonapi

Reply:
All API signatures are now JSON API standard.
But Rubocop is raising some warnings (that seems "false positive" IMO):
![image](https://user-images.githubusercontent.com/77609814/122384159-9e6dba00-cf95-11eb-9e83-9b11f50bcb76.png)

But anyway, it's the time for me to ask: Any other way to be JSON compliant with Doorkeeper::TokenController (without rewritting the whole controller?) :) 

Also I will request access to the postman you sent me: From the source code I cannot believe you have JSON:API valid error messages :P And I do not see where are tokens form handled (maybe I miss a Doorkeeper config?)

Reply:
Update: 
- I've put that code in a generic concern and check action names in the filter method itself. 
- Also I have realized that the requests tests were still executed on the Doorkeeper::TokenController (instead of API::V1::TokenController) so I've updated it as well.

But I'm still not understanding 100% how Survey app handle JsonAPI formatting (this app has much cleaner code than my solution so I'd be delighted to know more about it :) )
Thanks!

PS: Talking about 201 code, maybe the team wants to update this action with a 201 status code too :D
![image](https://user-images.githubusercontent.com/77609814/122495874-04e5ed00-d015-11eb-8a06-8833559fd130.png)
(as the action "creates a token", it does not log the user technically speaking)

Reply:
> But I'm still not understanding 100% how Survey app handles JsonAPI formatting 

@malparty Doorkeeper allows to customize token response without overriding everything in the its own token controller.([Customize success token response](https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-Token-Response), [Customize error response](https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-token-error-response)). In the Nimble Survey PR we did exactly the same thing in this PR you can check it https://github.com/nimblehq/nimble-survey-web/pull/14

> PS: Talking about 201 code, maybe the team wants to update this action with a 201 status code too :D

In the login/token API (`oath/token`), technically it creates a a token in the database, but from the client's perspective(or API perpective)  we are not creating any resource, we just want to login(or just getting a token from server). That why I think Doorkeeper  returns `201` status code for `oath/token` API.  

And to discussion with team and finalizing some conventions, we create a RFC(Request for comment) ticket in the compass, we vote there and update the conventions in the compass. For example, we created [this RFC](https://github.com/nimblehq/compass/issues/642) to finalize Go convention variable convention. You can create a RFC issue for 201/200 status code for `oauth/token` API if you want 👍

Reply:
Thanks a lot for awesome clear reply :) 
I will implement the changes now ~ 

For RFC 200/201: well, I believe it's not really important, as you said, the client can either ask "please create me a temporary access_token" or "please give me a way to log in" ^^, 
So I will follow Doorkeeper standard to not add useless config/overwrite ;-)